src/abc_atlas_access/abc_atlas_cache/manifest.py

```diff
@@ -0,0 +1,284 @@
+
+        Returns
+        -------
+        list of str
+            List of all data files that are not metadata in the specified
+            directory.
+        """
+        return self._list_data_in_directory(directory, is_metadata=False)
+
+    def get_file_attributes(
+            self,
+            directory: str,
+            file_name: str,
+    ) -> CacheFileAttributes:
+        """
+        Get the data file with the specified name in the specified directory.
+
+        Parameters
+        ----------
+        directory: str
+            The directory to look for the file in.
+        file_name: str
+             The name of the file to look for.
+
+        Returns
+        -------
+        CacheFileAttributes
+            The file attributes for the requested file.
+        """
+        file_name = file_name.split('/')
+        if len(file_name) == 1:
+            file_name = file_name[0]
+            kind = None
+        else:
+            kind = file_name[1]
+            file_name = file_name[0]
+        directory_data = self._data['file_listing'][directory]
+
+        file_attributes = None
+        for sub_dir in directory_data.keys():
+            if file_name in directory_data[sub_dir].keys():
+                files_data = directory_data[sub_dir][file_name]
+                if "files" in files_data.keys():
+                    file_type = list(files_data["files"].keys())[0]
+                    file_attributes = self._create_file_attributes(
+                        remote_path=files_data["files"][file_type]['url'],
+                        version=files_data["files"][file_type]['version'],
+                        size=files_data["files"][file_type]['size'],
+                        relative_path=files_data["files"][file_type][
+                            'relative_path'],
+                        file_type=file_type
+                    )
+                elif kind in files_data.keys():
+                    file_type = list(files_data[kind]['files'].keys())[0]
+                    file_attributes = self._create_file_attributes(
+                        remote_path=files_data[kind]["files"][file_type][
+                            'url'],
+                        version=files_data[kind]["files"][file_type][
+                            'version'],
+                        size=files_data[kind]["files"][file_type]['size'],
+                        relative_path=files_data[kind]["files"][file_type][
+                            'relative_path'],
+                        file_type=file_type
+                    )
+                elif kind is None and "files" not in files_data.keys():
+                    raise KeyError(
+                        f"File {file_name} found in directory but multiple "
+                        f"files found: {list(files_data.keys())}. Please "
+                        "specify the file name as one of "
+                        f"{['%s/%s' % (file_name, key) for key in files_data.keys()]}"
```

Comment:
Is this list going to be extremely long? If so, should the error message just include the first ~ 3 so that users can see the general form of the specification? The error message could then refer the user to a method that actually lists the available `(file_name, key)` pairs.

Feel free to disregard this if the list is not actually as long as I fear it will be.

Reply:
Currently, this is only 2 entries long. I suppose the answer to this is if we for some reason end up releasing files that are the same data but reprocessed in a different way. This switch here is for the log2/raw versions of the expression matrix files.

Reply:
Okay. I couldn't tell whether this was at the `log2` vs `raw` level or somewhere higher in the directory structures. Leave it as-is.